### PR TITLE
fix: clear readyQueue with dispose

### PR DESCRIPTION
**Pull request by brandonocasey**
## Description
Seems like we never clear the internal `readyQueue_` when calling `dispose`  which may cause the issues seen in https://github.com/videojs/video.js/pull/6701#issuecomment-736046505 and #5706

### CODE REVIEW

**Review by gkatsev — APPROVED** [review]

**Comment by brandonocasey** [comment]
I updated this pull request slightly when adding a unit test I realized that this case will also throw an error when it happens about `nodeName` since we don't check for null in `validateTarget`. The error was also too vague for my liking but we can back that part of the changes out if we think that it's too much for this pull request.

File: src/js/mixins/evented.js [hunk 1]
```diff
@@ -68,9 +88,9 @@ const isValidEventType = (type) =>
  * @param  {Object} target
  *         The object to test.
  */
-const validateTarget = (target) => {
-  if (!target.nodeName && !isEvented(target)) {
-    throw new Error('Invalid target; must be a DOM node or evented object.');
+const validateTarget = (target, obj, fnName) => {
```
Comment (brandonocasey) [inline src/js/mixins/evented.js:91]: Made these errors more specific.

File: src/js/mixins/evented.js [hunk 2]
```diff
@@ -68,9 +88,9 @@ const isValidEventType = (type) =>
...
+  if (!target || (!target.nodeName && !isEvented(target))) {
```
Comment (brandonocasey) [inline src/js/mixins/evented.js:92]: Check if target is null before trying to check a property on it.

File: src/js/mixins/evented.js [hunk 3]
```diff
@@ -398,6 +418,8 @@ const EventedMixin = {
    *          Whether or not the default behavior was prevented.
    */
   trigger(event, hash) {
+    validateTarget(this.eventBusEl_, this, 'trigger');
```
Comment (brandonocasey) [inline src/js/mixins/evented.js:421]: validate on trigger, this is likely the cause of the issue we see in `dom`

**Comment by gkatsev** [comment]
If it can be split out into two PRs, that would be nice. The changes look good.

**Comment by brandonocasey** [comment]
Moved the evented changes to https://github.com/videojs/video.js/pull/6982

**Review by gkatsev — APPROVED** [review]

**Pull request merged**